vault/dendron.tutorial.linking-notes.md

```diff
@@ -2,43 +2,102 @@
+
+![[Local Note Graph|dendron://dendron.dendron-site/dendron.topic.graph-view#full-note-graph:#*]]
+
+## Refactoring in Dendron
+
+You may wonder what happens to your links if you want to change the name of a note or header. Dendron keeps this in mind and provides commands to update your notes. All the while, Dendron updates any all wikilinks in your notes.
+
+### Rename Note
+
+The [[Rename Note|dendron://dendron.dendron-site/dendron.topic.refactoring#rename-note]] command is used to change the name of, and all links to, a single note.
+
+> 🌱 **ACTION:** Use the lookup to navigate to `[[recipes.italian.desserts.tiramisu]]`. Then:
+> - Run the command `Dendron: Rename Note` to change the title to `[[recipes.italian.desserts.gelato]]`.
+> - Use the lookup to navigate to `[[recipes.ingredients.shopping-list]]`. You should see the wikilink there has updated from `[[recipes.italian.desserts.tiramisu#ingredients]]` to `[[recipes.italian.desserts.gelato#ingredients]]`, ensuring links are updated!
+
+### Rename Header
+
+The [[Rename Header|dendron://dendron.dendron-site/dendron.topic.refactoring#rename-header]] command is used to change the name of, and all links to, a single header.
+
+> 🌱 **ACTION:** Use the lookup to navigate to `[[recipes.italian.desserts.gelato]]`. Then:
+> - Place the cursor in `## Ingredients`, and run the command `Dendron: Rename Note`
+> - Rename the note to `## Best Ingredients`
+> - Use the lookup to navigate to `[[recipes.ingredients.shopping-list]]`. You should see the wikilink there has updated from `[[recipes.italian.desserts.gelato#ingredients]]` to `[[recipes.italian.desserts.gelato#best-ingredients]]`, ensuring links are updated!
+
+If wanting to migrate a section beneath a certain header, to a different note entirely, [[Move Header|dendron://dendron.dendron-site/dendron.topic.refactoring#move-header]] will do the job and ensure links are updated.
+
+### Refactor Hierarchy
+
+The [[Refactor Hierarchy|dendron.topic.refactoring#refactor-hierarchy]] command is used to change the name of, and all links to, a collection of notes.
+
+> 🌱 **ACTION:** This section will be all about creating and refactoring notes, in order to get an idea for how it all works.
+
+- Create the following notes by placing the cursor inside each, and pressing `Ctrl+Enter`:
+  - `[[recipes.vegetarian.tikka-masala]]`
+  - `[[recipes.vegetarian.aloo-paratha]]`
+  - `[[recipes.vegetarian.dosa]]`
+- Each of these are indian food! Let's move them into the `indian` hierarchy.
+  - Run the command `Dendron: Refactor Hierarchy`
+  - At the first prompt, enter `recipes.vegetarian` (note hierarchy you are targeting)
+  - At the second prompt, enter `recipes.indian` (renaming the targeted hierarchy)
+
+A refactor preview should be provided, before making the change, to confirm the changes you are about to make.
+
+![[dendron://dendron.dendron-site/asset.preview#refactor-hierarchy-preview,1:#*]]
```

Comment:
heads up that the `# :*` syntax doesn't work if there is no last header on the page. you can fix this by adding a dummy header in the target (see example below)

```
### Refactor Hierarchy Preview

![Refactor Hierarchy Preview prompt](https://org-dendron-public-assets.s3.amazonaws.com/images/refactor-hierarchy-preview-prompt.png)

##
```

Reply:
I didn't know about this! This makes sense for why I've seen empty headers at the bottom of notes. I've fixed in another place where I was about to break that functionality since I was trying to clean up places.